File: bindings/java/examples/Sign257Example.java
import ua.dstucrypto.dstucore.Sign257;

/** {@code crypto_sign257} (DSTU 4145 {@code m=257}, T-199/T-204): generate a signing keypair,
 * sign a message, verify it. */
final class Sign257Example {
    private Sign257Example() {
    }

    static void run() throws Exception {
        byte[] signingKey = Sign257.keygen();
        byte[] verifyingKey = Sign257.verifyingKey(signingKey);

        byte[] message = "a message whose origin and integrity matter".getBytes("UTF-8");
        byte[] signature = Sign257.sign(signingKey, message);
        if (!Sign257.verify(verifyingKey, message, signature)) {
            throw new IllegalStateException("verification failed");
        }
        System.out.println("signed and verified a " + message.length + "-byte message");

        if (!Sign257.verify(verifyingKey, "a different message".getBytes("UTF-8"), signature)) {
            System.out.println("signature over a different message correctly rejected");
        }
    }
}

File: bindings/dotnet/examples/Sign257Example.cs
using System.Text;
using DstuCore;

namespace DstuCore.Examples;

/// <summary><c>crypto_sign257</c> (DSTU 4145 `m=257`, T-199/T-204): generate a signing keypair,
/// sign a message, verify it.</summary>
internal static class Sign257Example
{
    public static void Run()
    {
        using var signingKey = SigningKey257.Generate();
        using var verifyingKey = signingKey.VerifyingKey();

        var message = Encoding.ASCII.GetBytes("a message whose origin and integrity matter");
        var signature = signingKey.Sign(message);
        if (!verifyingKey.Verify(message, signature))
        {
            throw new InvalidOperationException("verification failed");
        }

        Console.WriteLine($"signed and verified a {message.Length}-byte message");

        if (!verifyingKey.Verify(Encoding.ASCII.GetBytes("a different message"), signature))
        {
            Console.WriteLine("signature over a different message correctly rejected");
        }
    }
}
